tendermint/src/lite/error.rs

```diff
@@ -63,6 +57,9 @@ pub enum Kind {
 
     #[error("A valid threshold is `1/3 <= threshold <= 1`, got: {got}")]
     InvalidTrustThreshold { got: String },
+
+    #[error("Implementation specific error: {0}")]
+    ImplementationSpecific(String),
```

Comment:
This isn't optimal in my eyes. Would be cool if we wouldn't  have to deal with strings here.

Reply:
This will invite lazy catch all for implementations. What control flow is desired for consumers?

Reply:
That's a very good point. In all other cases the verifier produces errors and consumers (e.g the light node implementation) can match over those and handle them as they wish. Here, roles are kinda reversed because these are errors that come from an implementation. Maybe these should rather be std errors instead. I'm currently looking into using the `context` method to see if we can decorate  `ImplementationSpecific ` with the original error (which doesn't need to be a String / could be anything which implements `Into<BoxError>`. Then, we have the same flexibility and users can get the source error back (and log it or do what they want with it).

https://github.com/interchainio/tendermint-rs/blob/ca6a17a4a9e4d6c65bdb6a19ccf069237919ad6e/tendermint/src/lite/error.rs#L68-L73

Reply:
That actually works! Had to modify rpc error to implement std::error:Error and I think we are good now.